staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -644,7 +686,13 @@ func (c *Cacher) GetList(ctx context.Context, key string, opts storage.ListOptio
 	trace := utiltrace.New("cacher list", utiltrace.Field{Key: "type", Value: c.objectType.String()})
 	defer trace.LogIfLong(500 * time.Millisecond)
 
-	c.ready.wait()
+	select {
+	case <-c.readyCh:
+	case <-ctx.Done():
+		return ctx.Err()
```

Comment:
the context cancellation is racy, if `isReady` but `context is cancelled`

Reply:
I'll add a comment, the context can be cancelled here or later in that case, this only handles the case when 
"is waiting for ready but not stopped" and the context is cancelled